addon/utils/load-relationship.js

```diff
@@ -11,14 +11,15 @@ export default function loadAll(model, relationship, dest, options = {}) {
 
     return model.query(relationship, query).then(results => {
         dest.pushObjects(results.toArray());
-
-        var total = results.meta.pagination.total;
-        var pageSize = results.meta.pagination.per_page;
-        var remaining = total - (page * pageSize);
-        if (remaining > 0) {
-            query.page = page + 1;
-            query['page[size]'] = pageSize;
-            return loadAll(model, relationship, dest, query);
+        if (results.meta) {
+            var total = results.meta.pagination.total;
+            var pageSize = results.meta.pagination.per_page;
```

Comment:
Page size changes through the API pass through to future pages. So https://api.osf.io/v2/nodes/?page[size]=5 will return pagination links like https://api.osf.io/v2/nodes/?page=2&page%5Bsize%5D=5